python-frontend/src/main/java/org/sonar/python/caching/CacheContextImpl.java

```diff
@@ -58,19 +60,24 @@ public PythonWriteCache getWriteCache() {
     return writeCache;
   }
 
-  public static CacheContextImpl of(SensorContext context) {
+  public static CacheContextImpl of(SensorContext context, @Nullable SonarLintCache sonarLintCache) {
     String sonarModules = context.config().get("sonar.modules").orElse("");
     boolean isUsingSonarModules = !sonarModules.isEmpty();
     if (isUsingSonarModules && context.isCacheEnabled()) {
       LOG.info("Caching will be disabled for this analysis due to the use of the \"sonar.modules\" property.");
     }
-    if (!context.runtime().getProduct().equals(SonarProduct.SONARLINT)
-      && context.runtime().getApiVersion().isGreaterThanOrEqual(MINIMUM_RUNTIME_VERSION)
-      && !isUsingSonarModules
-    ) {
-      return new CacheContextImpl(context.isCacheEnabled(), new PythonWriteCacheImpl(context.nextCache()), new PythonReadCacheImpl(context.previousCache()));
+
+    if (context.runtime().getApiVersion().isGreaterThanOrEqual(MINIMUM_RUNTIME_VERSION) &&
+      !isUsingSonarModules) {
+      if (context.runtime().getProduct().equals(SonarProduct.SONARLINT)) {
+        if (sonarLintCache != null) {
+          return new CacheContextImpl(true, new PythonWriteCacheImpl(sonarLintCache), new PythonReadCacheImpl(sonarLintCache));
```

Comment:
Until now, the `CacheContext#isCacheEnabled` was understood to mean that the SQ/SC PR analysis cache was enabled.

This information is then used by the `SonarQubePythonIndexer` to try to retrieve/store information from the PR analysis cache. I believe it is used similarly by the SonarSecurity frontend to retrieve and save UCFGs (not 100% sure about that - didn't double check).

By chance, none of these components are involved in the SonarLint context, but I'm afraid this is a bit implicit and could lead to nasty bugs should the underlying assumption change.

Even if we were setting `false` here, I feel the meaning of this field may no longer be clear, now that this component handles 2 distinct caches for 2 quite different use cases.